### PR TITLE
Added retry logic for suspend/resume operations on the Workflow run in unit test case.

### PR DESCRIPTION
**Pull request by sagarkapare**
Fix the flaky test - http://builds.cask.co/browse/CDAP-RUT15-18.
When workflow completes, its runtime info is removed. Suspending such workflow should result in 404. However there is slight delay in marking the Workflow as COMPLETED and removing its runtime info(program controller). Added retry logic for the suspend/resume operations in the test case.

### CODE REVIEW

**Comment by sagarkapare** [comment]
@bdmogal please take a look. thanks!

**Comment by sagarkapare** [comment]
build is running here - http://builds.cask.co/browse/CDAP-RUT76-1

File: cdap-app-fabric-tests/src/test/java/co/cask/cdap/internal/app/services/http/handlers/WorkflowHttpHandlerTest.java [hunk 1]
```diff
@@ -121,12 +121,22 @@ private Integer runningProgramCount(Id.Program program, String runId) throws Exc
...
+  private void waitForStatusCode(final String path, final Id.Program program, final int expectedStatusCode)
+    throws Exception {
+    Tasks.waitFor(true, new Callable<Boolean>() {
+      @Override
+      public Boolean call() throws Exception {
+        HttpResponse response = doPost(getVersionedAPIPath(path, Constants.Gateway.API_VERSION_3_TOKEN,
+                                                           program.getNamespaceId()));
+        return expectedStatusCode == response.getStatusLine().getStatusCode();
+      }
+    }, 180, TimeUnit.SECONDS);
```
Comment (bdmogal) [inline cdap-app-fabric-tests/src/test/java/co/cask/cdap/internal/app/services/http/handlers/WorkflowHttpHandlerTest.java]: do we need such a long delay?

**Comment by bdmogal** [comment]
@sagarkapare how do we ensure that the workflow completes in these tests? Should that wait for the RunRecord and the ProgramControllerState to be marked completed? Just thinking that this will only fix suspend/resume but the other option may fix other edge cases?

File: cdap-app-fabric-tests/src/test/java/co/cask/cdap/internal/app/services/http/handlers/WorkflowHttpHandlerTest.java (hunk 1 above)
Comment (sagarkapare) [inline cdap-app-fabric-tests/src/test/java/co/cask/cdap/internal/app/services/http/handlers/WorkflowHttpHandlerTest.java]: Its worst case delay. Ideally it should happen in couple of retries.

**Comment by sagarkapare** [comment]
There is a separate check for the run completion. For example - https://github.com/caskdata/cdap/blob/release/3.5/cdap-app-fabric-tests/src/test/java/co/cask/cdap/internal/app/services/http/handlers/WorkflowHttpHandlerTest.java#L414

File: cdap-app-fabric-tests/src/test/java/co/cask/cdap/internal/app/services/http/handlers/WorkflowHttpHandlerTest.java (hunk 1 above)
Comment (bdmogal) [inline cdap-app-fabric-tests/src/test/java/co/cask/cdap/internal/app/services/http/handlers/WorkflowHttpHandlerTest.java]: Let's reduce it then. If there's something wrong in the environment, it will unnecessarily keep waiting for 3 min otherwise. How about 10 seconds?

Comment (sagarkapare) [inline cdap-app-fabric-tests/src/test/java/co/cask/cdap/internal/app/services/http/handlers/WorkflowHttpHandlerTest.java]: Reduced to 60 seconds.

**Comment by sagarkapare** [comment]
Merging to unblock others.
